### ISSUE TITLE
Game freezes in the main menu (macOS Monterey 12.2, M1, ARM64)

### ISSUE THREAD

**Issue by ivxvm** [minetest/minetest#12031]
##### Minetest version
```
% /Applications/minetest.app/Contents/MacOS/minetest --version

Minetest 5.6.0-dev-128f635 (OSX)
Using Irrlicht 1.9.0mt4
Using Lua 5.1.5
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_CURL=1
USE_GETTEXT=0
USE_SOUND=1
STATIC_SHAREDIR="minetest.app/Contents/Resources"
```

##### OS / Hardware
```
Operating system: macOS Monterey 12.2
CPU: Apple M1
```

##### Summary
Game freezes in the main menu, UI doesn't respond to anything.

##### Steps to reproduce
Build app bundle from source, sign using codesign, copy to /Applications, run

**Comment by sfan5** [comment]
Do you have access to a suitable debugger?
Under Linux I'd go about it this way:
* Run the game under gdb (`gdb /path/to/minetest`)
* `r` to start it
* when it freezes interrupt it using Ctrl+C
* type `thread apply all bt` to find where it's hanging

**Comment by ivxvm** [comment]
@sfan5 
Seems like gdb isn't supported on M1, but there is [lldb](https://lldb.llvm.org/use/map.html):

```
% lldb /Applications/minetest.app/Contents/MacOS/minetest 
(lldb) target create "/Applications/minetest.app/Contents/MacOS/minetest"
Current executable set to '/Applications/minetest.app/Contents/MacOS/minetest' (arm64).



(lldb) r
Process 34083 launched: '/Applications/minetest.app/Contents/MacOS/minetest' (arm64)
2022-01-31 22:17:45.365594+0200 minetest[34083:393811] [plugin] AddInstanceForFactory: No factory registered for id <CFUUID 0x102d9d400> F8BB1C28-BAE8-11D6-9C31-00039315CD46
Process 34083 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGTERM
    frame #0: 0x00000001816a0ebc libsystem_kernel.dylib`__semwait_signal + 8
libsystem_kernel.dylib`__semwait_signal:
->  0x1816a0ebc <+8>:  b.lo   0x1816a0edc               ; <+40>
    0x1816a0ec0 <+12>: pacibsp 
    0x1816a0ec4 <+16>: stp    x29, x30, [sp, #-0x10]!
    0x1816a0ec8 <+20>: mov    x29, sp


(lldb) bt all
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGTERM
  * frame #0: 0x00000001816a0ebc libsystem_kernel.dylib`__semwait_signal + 8
    frame #1: 0x00000001815abd88 libsystem_c.dylib`nanosleep + 216
    frame #2: 0x00000001003d8484 minetest`irr::CIrrDeviceMacOSX::sleep(unsigned int, bool) + 168
    frame #3: 0x00000001000dd8e8 minetest`GUIEngine::run() + 1028
    frame #4: 0x00000001000dccac minetest`GUIEngine::GUIEngine(JoystickController*, irr::gui::IGUIElement*, RenderingEngine*, IMenuManager*, MainMenuData*, bool&) + 1120
    frame #5: 0x0000000100027af4 minetest`ClientLauncher::main_menu(MainMenuData*) + 528
    frame #6: 0x0000000100026c78 minetest`ClientLauncher::launch_game(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&, bool, GameStartData&, Settings const&) + 1364
    frame #7: 0x0000000100024f80 minetest`ClientLauncher::run(GameStartData&, Settings const&) + 2076
    frame #8: 0x00000001002c341c minetest`main + 8760
    frame #9: 0x00000001007290f4 dyld`start + 520
  thread #2
    frame #0: 0x000000018169f604 libsystem_kernel.dylib`__workq_kernreturn + 8
  thread #3
    frame #0: 0x000000018169f604 libsystem_kernel.dylib`__workq_kernreturn + 8
  thread #4, name = 'AMCP Logging Spool'
    frame #0: 0x000000018169d990 libsystem_kernel.dylib`semaphore_wait_trap + 8
    frame #1: 0x000000018a0ac8ac caulk`caulk::mach::semaphore::wait_or_error() + 28
    frame #2: 0x000000018a08f65c caulk`caulk::concurrent::details::worker_thread::run() + 56
    frame #3: 0x000000018a08f27c caulk`void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*> > >(void*) + 96
    frame #4: 0x00000001816d9240 libsystem_pthread.dylib`_pthread_start + 148
  thread #5, name = 'com.apple.audio.IOThread.client'
    frame #0: 0x000000018169d954 libsystem_kernel.dylib`mach_msg_trap + 8
    frame #1: 0x000000018169dd00 libsystem_kernel.dylib`mach_msg + 76
    frame #2: 0x00000001833dfef0 CoreAudio`HALB_MachPort::SendSimpleMessageWithSimpleReply(unsigned int, unsigned int, int, int&, bool, unsigned int) + 104
    frame #3: 0x000000018326ee04 CoreAudio`HALC_ProxyIOContext::IOWorkLoop() + 3360
    frame #4: 0x000000018326dae0 CoreAudio`invocation function for block in HALC_ProxyIOContext::HALC_ProxyIOContext(unsigned int, unsigned int) + 100
    frame #5: 0x0000000183439420 CoreAudio`HALB_IOThread::Entry(void*) + 88
    frame #6: 0x00000001816d9240 libsystem_pthread.dylib`_pthread_start + 148
  thread #7
    frame #0: 0x000000018169f604 libsystem_kernel.dylib`__workq_kernreturn + 8
  thread #8
    frame #0: 0x000000018169f604 libsystem_kernel.dylib`__workq_kernreturn + 8
  thread #9, name = 'com.apple.NSEventThread'
    frame #0: 0x000000018169d954 libsystem_kernel.dylib`mach_msg_trap + 8
    frame #1: 0x000000018169dd00 libsystem_kernel.dylib`mach_msg + 76
    frame #2: 0x00000001817a4ed8 CoreFoundation`__CFRunLoopServiceMachPort + 372
    frame #3: 0x00000001817a3390 CoreFoundation`__CFRunLoopRun + 1212
    frame #4: 0x00000001817a2734 CoreFoundation`CFRunLoopRunSpecific + 600
    frame #5: 0x0000000184465fb0 AppKit`_NSEventThread + 196
    frame #6: 0x00000001816d9240 libsystem_pthread.dylib`_pthread_start + 148
  thread #10, name = 'AsyncWorker-0'
    frame #0: 0x000000018169d990 libsystem_kernel.dylib`semaphore_wait_trap + 8
    frame #1: 0x0000000100186a2c minetest`AsyncEngine::getJob(LuaJobInfo*) + 40
    frame #2: 0x0000000100187470 minetest`AsyncWorkerThread::run() + 384
    frame #3: 0x00000001001eb8a8 minetest`Thread::threadProc(Thread*) + 104
    frame #4: 0x00000001001ebaf4 minetest`void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(Thread*), Thread*> >(void*) + 44
    frame #5: 0x00000001816d9240 libsystem_pthread.dylib`_pthread_start + 148
  thread #11, name = 'AsyncWorker-1'
    frame #0: 0x000000018169d990 libsystem_kernel.dylib`semaphore_wait_trap + 8
    frame #1: 0x0000000100186a2c minetest`AsyncEngine::getJob(LuaJobInfo*) + 40
    frame #2: 0x0000000100187470 minetest`AsyncWorkerThread::run() + 384
    frame #3: 0x00000001001eb8a8 minetest`Thread::threadProc(Thread*) + 104
    frame #4: 0x00000001001ebaf4 minetest`void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(Thread*), Thread*> >(void*) + 44
    frame #5: 0x00000001816d9240 libsystem_pthread.dylib`_pthread_start + 148
  thread #12, name = 'AsyncWorker-2'
    frame #0: 0x000000018169d990 libsystem_kernel.dylib`semaphore_wait_trap + 8
    frame #1: 0x0000000100186a2c minetest`AsyncEngine::getJob(LuaJobInfo*) + 40
    frame #2: 0x0000000100187470 minetest`AsyncWorkerThread::run() + 384
    frame #3: 0x00000001001eb8a8 minetest`Thread::threadProc(Thread*) + 104
    frame #4: 0x00000001001ebaf4 minetest`void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(Thread*), Thread*> >(void*) + 44
    frame #5: 0x00000001816d9240 libsystem_pthread.dylib`_pthread_start + 148
  thread #13, name = 'AsyncWorker-3'
    frame #0: 0x000000018169d990 libsystem_kernel.dylib`semaphore_wait_trap + 8
    frame #1: 0x0000000100186a2c minetest`AsyncEngine::getJob(LuaJobInfo*) + 40
    frame #2: 0x0000000100187470 minetest`AsyncWorkerThread::run() + 384
    frame #3: 0x00000001001eb8a8 minetest`Thread::threadProc(Thread*) + 104
    frame #4: 0x00000001001ebaf4 minetest`void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(Thread*), Thread*> >(void*) + 44
    frame #5: 0x00000001816d9240 libsystem_pthread.dylib`_pthread_start + 148
(lldb) 
```

**Comment by sfan5** [comment]
Might be fixed by 951604e29ff9d4b796003264574e06031c014a3f.

**Comment by TochigiV** [comment]
> Might be fixed by [951604e](https://github.com/minetest/minetest/commit/951604e29ff9d4b796003264574e06031c014a3f).

Seems to be working for me. Thanks!
